backend/app/services/search.py

```diff
@@ -1,42 +1,222 @@
+        url = config.get(path+'.url')
+        esUser = None
         if index == "":
-            self.indice = self.cfg.get(configpath+'.indice')
+            indice = config.get(path+'.indice')
         else:
-            self.indice = index
-        if self.cfg.is_set(configpath+'.username') and \
-                self.cfg.is_set(configpath+'.password'):
-            self.esUser = self.cfg.get(configpath+'.username')
-            self.esPass = self.cfg.get(configpath+'.password')
-        if self.esUser :
-            self.es = AsyncElasticsearch(
-                    self.url,
+            indice = index
+        if config.is_set(path+'.username') and \
+                config.is_set(path+'.password'):
+            esUser = config.get(path+'.username')
+            esPass = config.get(path+'.password')
+        if esUser :
+            es = AsyncElasticsearch(
+                    url,
                     use_ssl=False,
                     verify_certs=False,
-                    http_auth=(self.esUser,self.esPass)
+                    http_auth=(esUser,esPass)
             )
         else:
-            self.es = AsyncElasticsearch(self.url, verify_certs=False)
+            es = AsyncElasticsearch(url, verify_certs=False)
+        return es, indice
+
+    def __init__(self,configpath="",index=""):
+        cfg = config.get_config()
+        self.new_es, self.new_index = self.initialize_es(cfg, configpath, index)
+        self.prev_es = None
+        if cfg.get(configpath + '.internal'):
+            self.prev_es, self.prev_index = self.initialize_es(cfg, configpath + '.internal', index)
+    
+    async def scan_indices(self, es_client, indice, query, timestamp_field, start_date, end_date, size):
+        indices = await self.get_indices_from_alias(es_client, indice)
+        if not indices:
+            indices = [indice]
+        sorted_index_list = SortedIndexList()
+        for index in indices:
+            sorted_index_list.insert(IndexTimestamp(index, await self.get_timestamps(es_client, index, timestamp_field)))
+        filtered_indices = sorted_index_list.get_indices_in_given_range(start_date, end_date)
+        results = []
+        for each_index in filtered_indices:
+            query['query']['bool']['filter']['range'][timestamp_field]['lte'] = str(min(end_date, each_index.timestamps[1]))
+            query['query']['bool']['filter']['range'][timestamp_field]['gte'] = str(max(start_date, each_index.timestamps[0]))
+            response = await es_client.search(
+                index=each_index.index,
+                body=jsonable_encoder(query),
+                size=size)
+            results.extend(response['hits']['hits'])
+        return results
+
+    async def post(self, query, indice=None, size=10000, start_date=None, end_date=None, timestamp_field=None):
+        if size == 0:
+            if self.prev_es:
+                self.prev_index = 'ospst-' + self.prev_index if indice is None else indice
```

Comment:
can we create a index-prefix var vs hard-setting to ospst which is specific to our OCP team. The Dashboard has many teams integrated now.